pythonbits/imagehosting.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+from .config import config
+
+from .imgur import ImgurUploader
+from .ptpimg import PtpImgUploader
+
+config.register('ImageHosting', 'provider',
+                "Enter a provider for image hosting, supported options are "
+                "ptpimg or imgur",
+                ask=True)
+
+
+def upload_files(images):
+    provider = config.get('ImageHosting', 'provider')
+    if provider.lower() == 'imgur':
+        return ImgurUploader().upload(images)
+    elif provider.lower() == 'ptpimg':
+        return PtpImgUploader().upload_files(*images)
+
```

Comment:
The cleaner approach would of course to have a ImageUplader base class, but this can be added later

Reply:
Yeah this definitely feels hacky, it should probably be replaced with dependency_injector or something.